### PR TITLE
Do a bunch of cleanup

### PR DESCRIPTION
**Pull request by yuvipanda**
- Code cleanup, I love removing lines of code!
- Allows use with binder more easily, you can specify a start URL of /rstudio

### CODE REVIEW

File: nbrsessionproxy/handlers.py [hunk 1]
```diff
@@ -67,112 +54,50 @@ class RSessionProxyHandler(IPythonHandler):
...
-        return '{}proxy/{}/'.format(self.base_url, self.state['port'])
-
-    def gen_response(self, proc):
-        response = {
-            'pid': proc.pid,
-            'url':self.rsession_uri(),
-        }
-        return response
-        
-    def get_client_id(self):
```
Comment (ryanlovett) [inline nbrsessionproxy/handlers.py]: Now that nbserverproxy does auth right, I guess get_client_id() won't be necessary! It wasn't being used, but I thought it'd be be necessary for security on multi-user servers.

File: nbrsessionproxy/handlers.py [hunk 2]
```diff
@@ -67,112 +54,50 @@ class RSessionProxyHandler(IPythonHandler):
...
-        if not clientid:
-            return False
 
-        uri = self.rsession_uri()
-        
-        
+    @gen.coroutine
     @web.authenticated
-    def post(self):
+    def get(self):
```
Comment (ryanlovett) [inline nbrsessionproxy/handlers.py]: I thought POST was the correct verb here since we're creating something.

File: nbrsessionproxy/handlers.py [hunk 3]
```diff
@@ -185,61 +110,28 @@ def post(self):
...
-        response = self.gen_response(proc)
 
-        client_id = self.get_client_id()
-        self.log.debug('post: client_id: {}'.format(client_id))
-        self.finish(json.dumps(response))
+        for i in range(5):
+            if (yield self.is_running()):
+                self.log.info('rsession startup complete')
+                break
+            # Simple exponential backoff
```
Comment (ryanlovett) [inline nbrsessionproxy/handlers.py]: Looks familiar. :)

Comment (ryanlovett) [reply nbrsessionproxy/handlers.py]: (the exponential backoff)

File: nbrsessionproxy/handlers.py [hunk 4]
```diff
@@ -67,17 +58,18 @@ class RSessionProxyHandler(IPythonHandler):
 
     def initialize(self, state):
         self.state = state
+        self.port = 9797
```
Comment (ryanlovett) [inline nbrsessionproxy/handlers.py]: This prevents more than one user from running this from non-container spawners.

Comment (yuvipanda) [reply nbrsessionproxy/handlers.py]: Yup, good catch. Updated to be dynamic again, but a little cleaner.

**Comment by yuvipanda** [comment]
@ryanlovett re: POST vs GET, I'd agree that POST is sortof the better technically correct verb to use. However, making it a GET means people can just go to a URL that shows them rstudio, similar to /tree or /lab, and I think that's worth it!

**Pull request merged**